screenpy_examples/screenpy/quietly_logging/actions/do_thing.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from screenpy import IsEqual, Performable, beat
+
+from screenpy_examples.screenpy.quietly_logging.actions.see import See
+from screenpy_examples.screenpy.quietly_logging.questions import SimpleQuestion
+
+if TYPE_CHECKING:
+    from screenpy import Actor
+
+
+class DoPassAfterAWhile(Performable):
+    LIMIT = 5
+    COUNTER = 0
+
+    @beat("{} tries to DoPassAfterAWhile")
+    def perform_as(self, actor: Actor):
+        if DoPassAfterAWhile.COUNTER > DoPassAfterAWhile.LIMIT:
+            DoPassAfterAWhile.COUNTER = 0
+            return
+
+        DoPassAfterAWhile.COUNTER += 1
```

Comment:
Editing the static class variable means that every instance of `DoPassAfterAWhile` will pass after that first one does, is this intentional?

Reply:
This is intentional.   It's only used in one test.  In this test we want to show how `Eventually` works.  The action needs to fail for the first few loops then start to pass.  Only way to do that is keep a counter that is outside the instance.

```python
def test_eventually_succeeds(marcel: Actor) -> None:
    marcel.will(Eventually(DoPassAfterAWhile()).for_(1).seconds())
```

Reply:
I take that back, it's not the ONLY way.  It's the only way when calling the actions normally.
`marcel.will(Eventually(PerformPassAfterAWhile()).for_(1).seconds())` could be replaced with an instance.  

But this requires a little more setup which felt like it would just complicate the example.
```python
instance_of_perform_pass_after_a_while = PerformPassAfterAWhile()
marcel.will(Eventually(instance_of_perform_pass_after_a_while).for_(1).seconds())
```